### ISSUE TITLE
[softmax] softmax perf regression between pytorch 2.9 and 2.10

### ISSUE THREAD

**Issue by weishi-deng** [intel/intel-xpu-backend-for-triton#5750]
### Describe the issue

triton_red_fused__softmax__to_copy__unsafe_view_add_permute_8 kernel gets 70% regression compared to pytoch 2.9 release.

### Environment details

old env:
pytorch 2.9.0 + pytorch_triton_xpu-3.5.0
reproducer:
pip install torch==2.9.0 torchvision torchaudio --index-url https://download.pytorch.org/whl/xpu
[softmax_old.py](https://github.com/user-attachments/files/24328574/softmax_old.py)

new env:
pytorch 2.10.0 + pytorch-triton-xpu-3.5.0+gitaa01f5c2
reproducer:
pip install torch==2.10.0 torchvision torchaudio --index-url https://download.pytorch.org/whl/test/xpu
[softmax_new.py](https://github.com/user-attachments/files/24328577/softmax_new.py)

**Comment by weishi-deng** [comment]
pytorch issue link: https://github.com/intel/torch-xpu-ops/issues/2619
